src/axum/serde_json.rs

```diff
@@ -0,0 +1,99 @@
+use std::marker::PhantomData;
+
+use crate::errors::JsonError;
+use crate::{DeserializeError, Deserr};
+use axum::extract::rejection::JsonRejection;
+use axum::extract::FromRequest;
+use axum::response::IntoResponse;
+use axum::Json;
+use http::StatusCode;
+
+/// Extractor for typed data from Json request payloads
+/// deserialised by deserr.
+///
+/// ## Extractor
+/// To extract typed data from a request body, the inner type `T` must implement the
+/// [`deserr::Deserr<E>`] trait. The inner type `E` must implement the
+/// [`DeserializeError`] trait.
+///
+/// ## Response
+/// [`axum::response::IntoResponse`] is implemented for any `AxumJson<T, E>`
+/// where `T` implement [`serde::Serialize`].
+#[derive(Debug)]
+pub struct AxumJson<T, E>(pub T, PhantomData<E>);
+
+impl<T, E> AxumJson<T, E> {
+    pub fn new(data: T) -> Self {
+        AxumJson(data, PhantomData)
+    }
+
+    pub fn into_inner(self) -> T {
+        self.0
+    }
+}
+
+#[derive(Debug)]
+pub enum AxumJsonRejection<E: DeserializeError> {
+    DeserrError(E),
+    JsonRejection(JsonRejection),
+}
+
+impl<E: DeserializeError + std::fmt::Display> std::fmt::Display for AxumJsonRejection<E> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            AxumJsonRejection::DeserrError(e) => e.fmt(f),
+            AxumJsonRejection::JsonRejection(e) => e.fmt(f),
+        }
+    }
+}
+
+impl<T, E> IntoResponse for AxumJson<T, E>
+where
+    T: serde::Serialize,
+{
+    fn into_response(self) -> axum::response::Response {
+        Json(self.0).into_response()
+    }
+}
```

Comment:
Hey, I don’t think deserr should be used to serialize data.
It doesn’t implement half of the features serde provides currently and doesn’t bring anything.